### DISCUSSION TITLE
Suggestion for a modifcation for a SaaS dashboard builder

### DISCUSSION THREAD

**Discussion by mehdiamenein** [Budibase/budibase#4104] [Help]
Dear all, 

I hope you are well,

First of all huge thanks to budibase team for developing this solution and are working on improving it actively, I have recently started playing around with it as I was thinking of using budibase as a dashboard for my SaaS.

I use Oauth2 flow for authentication of my applications and usually use Auth0 in the process, configuring it was a extremely easy and satisfying process on budibase so I could easily move my user base to a dashboard I develop with it.

However I have a some major problems, which I was hoping to get some thoughts on feasiblity of their implementation and if they are on the roadmap of the team.

1. If a user redirects to the auth0 page and registers there, he will be redirected back and won't have access to the application until I give him/her the access manually. Can there be a webhook in automation for creating a user? I mean like a global webhook not app scoped webhook, I tried creating user with automation but it only adds them the db preview of the app and not my user directory
1.1. Another option will be giving the possiblity of defining default app which everyone can get a default access to upon registeration 
2. The local user login (email and password) is not a desirable auth flow at all, isn't there any possiblity to remove local flow completely from the login/register page?
3. is there a possiblity to add a /?app_redirect={app_id} to the login page so that when the user is logged in instead of seeing the generic dashboard, they get redirected to the correct dashboard that they have access to?

Looking forward to hearing from you soon,

Bests,
Mehdi

**Comment by mjashanks** [comment]
Hi Medhi. Thanks, and welcome.

1. Right now, you always need to manually add a user to Budibase. You add them by their email address. Only then can they login with Auth0. There is no such thing as a global webhook right now. Over the next few months, we will be documenting the Budibase API - once this is available you should be able to create users by combine an automation with an API request to Budibase. The automation could just live in a separate app.
2. This has been asked for before - I agree - we should do this. Not sure when we will do it.
3. Again - good suggestion. Lots of people have asked for users to be automatically redirected to an app if they only have access to one app. We are currently working on a feature/fix that will allow the following
- Pretty URLs for apps: `https://yourdomain.com/app/my-app-name`, and allowing control over the app url
- When users aren't logged in, and access the app URL, they will get sent to the app, not the portal.
Is this enough to solve your issue?

**Reply by mehdiamenein** [reply]
Hello @mjashanks 

Thank you for your response,

Regarding point 1 => Is there anything I can do to speed this up? or alternatively is there maybe a dirty trick for now to make this possible, I remember back in Strapi 3 they had doc to created custom webhook api token functionality until they release a built-in solution for it, maybe with the help of external module as described here, this also be possible https://docs.budibase.com/references-1/contributing-to-budibase/contribute-data-connectors

Regarding point 2 => I actually went through the code and I might be able to fix this myself temporary however the onboarding seems a bit more time consuming than I initially thought for setting up a development environment, sorry if I am asking an obvious question but I could not find this in the contribution section of the doc, "Is there any developer on boarding documentations for someone who wants to add a small addition to the application?"

Regarding number 3 => I believe it should be fine

Also thank you for the update on the new feature/fix that is coming out, can't wait to see it

**My Proposal for Point 2**
I need to first take a look at the auth config database table but I believe there should be the possiblity of adding an additional activate option which then can be checked by a similar variable to "loaded" on https://github.com/Budibase/budibase/blob/develop/packages/builder/src/pages/builder/auth/login.svelte
![image](https://user-images.githubusercontent.com/73882408/150190755-03491ae1-6842-496e-a0e6-076f96cf6e02.png)
Then based on the fact that it is activated or deactivated it should be possible to hide this section
![image](https://user-images.githubusercontent.com/73882408/150190388-83908da1-e749-46f3-8331-1290547b989d.png)

If it is possible and you can lead me to some documentations and references I can use to adapt myself to the source code I will be glad to help out, I know you are already busy with implementing a lot of other things, so I do not want to put more pressure on your workload

Bests,
Mehdi

**Reply by mjashanks** [reply]
Hey Mehdi - a small correction to my answer above.

It turns out that you do not need to create the user manually, when using SSO. If a user log's in successfully using OIDC, they will get added as a user. However, they will be added with no access to apps, so there is still a manual step there for someone to give them access.

In future, we will be introducing the concept of "Groups" or "Workspaces", whereby you can give users default permissions based on their group membership in the Auth provider.
